bpf/lib/wireguard.h

```diff
@@ -65,31 +65,20 @@ wg_maybe_redirect_to_encrypt(struct __ctx_buff *ctx)
 	case bpf_htons(ETH_P_IP):
 		if (!revalidate_data(ctx, &data, &data_end, &ip4))
 			return DROP_INVALID;
+
 # if defined(TUNNEL_MODE)
-		/* A rudimentary check (inspired by is_enap()) whether a pkt
-		 * is coming from tunnel device. In tunneling mode WG needs to
-		 * encrypt such pkts, so that src sec ID can be transferred.
+		/* In tunneling mode WG needs to encrypt tunnel traffic,
+		 * so that src sec ID can be transferred.
 		 *
 		 * This also handles IPv6, as IPv6 pkts are encapsulated w/
 		 * IPv4 tunneling.
 		 */
-		if (ip4->protocol == IPPROTO_UDP) {
-			int l4_off = ETH_HLEN + ipv4_hdrlen(ip4);
-			__be16 dport;
-
-			if (l4_load_port(ctx, l4_off + UDP_DPORT_OFF, &dport) < 0) {
-				/* IP fragmentation is not expected after the
-				 * encap. So this is non-Cilium's pkt.
-				 */
-				break;
-			}
-
-			if (dport == bpf_htons(TUNNEL_PORT)) {
-				from_tunnel = true;
-				break;
-			}
+		if (ctx_is_overlay(ctx)) {
+			from_tunnel = true;
+			break;
```

Comment:
@cilium/wireguard I have a feeling we should only change wireguard in 1.17, after we can trust that `to-overlay` *will* mark all traffic as needed. Otherwise we might have a few unmarked packets in-flight, that would no longer get handled here.

Reply:
Split the wireguard part up into two parts - let's add the TODO now, and then do the conversion in v1.17.

(will drop the second patch before merging, just as preview what it will look like)

Reply:
alright, dropped the second patch. Let's do this.